### PR TITLE
Give the brief description of each parameter in site intent and workf…

### PR DESCRIPTION
**Pull request by Abhishek-121**
…low manager module and explain things in detail like type of rf_model in floor site etc.

In this commit -

— Give the brief description of each parameter in site intent and workflow manager module and explain things in detail like type of rf_model in floor site etc.

— Changed boolean value from true/false to True/False in both modules and site playbook.

-- Write API to convert site_type to type in site workflow manager and intent module and update type to site_type in playbook as well.

### CODE REVIEW

File: plugins/modules/site_intent.py [hunk 1]
```diff
@@ -35,50 +35,48 @@
...
+            description: Contains details for creating or managing a building within a site.
             type: dict
             suboptions:
               address:
                 description: Address of the building to be created.
                 type: str
               latitude:
-                description: Latitude coordinate of the building (eg 37.338).Values between -90 to +90.
-                type: int
+                description: Latitude coordinate of the building (eg 37.338). Values between -90 to +90.
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:68]: description: Geographical latitude coordinate of the building. For example, use 37.338 for a location in San Jose, California. Valid values range from -90.0 to +90.0 degrees. Type is int of float?

Also you removed the "type: float" Can you please add it?

File: plugins/modules/site_intent.py [hunk 2]
```diff
@@ -35,50 +35,48 @@
...
               longitude:
-                description: Longitude coordinate of the building (eg -121.832).Values between -180 to +180.
+                description: Longitude coordinate of the building (eg -121.832). Values between -180 to +180.
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:70]: description: Geographical longitude coordinate of the building. For example, use -121.832 for a location in San Jose, California. Valid values range from -180.0 to +180.0 degrees.
  type: float

File: plugins/modules/site_intent.py [hunk 3]
```diff
@@ -35,50 +35,48 @@
...
                 type: int
               name:
                 description: Name of the building (eg building1).
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:73]: description: Name of the building (e.g., "Building1").
or
  description: Name of the building, for example "Building1".

File: plugins/modules/site_intent.py [hunk 4]
```diff
@@ -35,50 +35,48 @@
...
                 type: str
               parent_name:
-                description: Complete Parent name of the Building to be created/deleted(eg Global/USA/San Francisco).
+                description: Complete parent name of the building to be created/deleted(eg Global/USA/San Francisco).
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:76]: description: Hierarchical parent path of the building, indicating its location within the site (e.g., "Global/USA/San Francisco").

File: plugins/modules/site_intent.py [hunk 5]
```diff
@@ -35,50 +35,48 @@
...
                 type: str
           floor:
-            description: Site Create's floor.
+            description: Contains details for creating or managing a floor within a site.
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:79]: description: Details required for creating or managing a floor within a site.
or
description: Configuration details required for creating or managing a floor within a site.

suboptions:
  height:
    description: Height of the floor in feet (e.g., 15).
    type: int
  length:
    description: Length of the floor in feet (e.g., 100).
    type: int


name:
  description: Name of the floor (e.g., "Floor-1").
  type: str

File: plugins/modules/site_intent.py [hunk 6]
```diff
@@ -90,18 +88,27 @@
               name:
                 description: Name of the floor (eg floor-1).
                 type: str
-              parentName:
-                description: Complete Parent name of the floor to be created(eg Global/USA/San Francisco/BGL_18).
+              parent_name:
+                description: Complete parent name of the floor to be created(eg Global/USA/San Francisco/BGL_18).
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:92]: description: Hierarchical parent path of the floor, indicating its location within the site (e.g., "Global/USA/San Francisco/BGL_18").

File: plugins/modules/site_intent.py [hunk 7]
```diff
@@ -90,18 +88,27 @@
...
                 type: str
               rf_model:
-                description: Type of floor. Allowed values are 'Cubes And Walled Offices',
-                  'Drywall Office Only', 'Indoor High Ceiling', 'Outdoor Open Space'.
+                description: Type of floor (allowed values are 'Cubes And Walled Offices', 'Drywall Office Only', 'Indoor High Ceiling',
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:95]: The RF (Radio Frequency) model type for the floor, which is essential for simulating and optimizing wireless network coverage. Select from the following allowed values, which describe different environmental signal propagation characteristics.

Then you can explain all the types

File: plugins/modules/site_intent.py [hunk 8]
```diff
@@ -90,18 +88,27 @@
...
+                        radio signals may experience attenuation due to walls and obstacles.
+                    Drywall Office Only - This RF model indicates an environment with drywall partitions, commonly found in office spaces,
+                        which may have moderate signal attenuation.
+                    Indoor High Ceiling - This RF model is suitable for indoor spaces with high ceilings, such as auditoriums or atriums,
+                        where signal propagation may differ due to the height of the ceiling.
+                    Outdoor Open Space - This RF model is used for outdoor areas with open spaces, where signal propagation is less obstructed
+                        and may follow different patterns compared to indoor environments.
                 type: str
               width:
                 description: Width of the floor units is ft. (eg 100).
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:108]: description: Width of the floor in feet (e.g., 100).

File: plugins/modules/site_intent.py [hunk 9]
```diff
@@ -90,18 +88,27 @@
...
                 type: int
               floor_number:
-                description: Floor number in the building/site (eg 5).once created, it can't be modified.
+                description: Floor number in the building/site (eg 5) can be given only while creating the floor site.
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:111]: description: Floor number within the building/site (e.g., 5). This value can only be specified during the creation of the floor and cannot be modified afterward.

File: plugins/modules/site_intent.py [hunk 10]
```diff
@@ -136,7 +143,7 @@
     - site:
         area:
           name: string
-          parentName: string
+          parent_name: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:146]: Can we give proper name?

File: plugins/modules/site_intent.py [hunk 11]
```diff
@@ -158,7 +165,7 @@
           latitude: 0
           longitude: 0
           name: string
-          parentName: string
+          parent_name: string
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:168]: Can we give a proper playbook?

File: plugins/modules/site_intent.py [hunk 12]
```diff
@@ -35,50 +35,48 @@
...
-    - List of details of site being managed.
+    description: It represents a list of details for creating/managing/deleting sites, including areas, buildings, and floors.
     type: list
     elements: dict
-    required: true
+    required: True
     suboptions:
       type:
-        description: Type of site to create/update/delete (eg area, building, floor).
+        description: Specifies the type of site operation to perform (e.g. create, update, delete).
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/site_intent.py]: Changed and write one function for converting site_type to type in site module.

File: plugins/modules/site_intent.py [hunk 13]
```diff
@@ -35,50 +35,48 @@
...
+        description: Contains details about the site being managed including areas, buildings and floors.
         type: dict
         suboptions:
           area:
-            description: Site Create's area.
+            description: Contains details for creating or managing an area within a site.
             type: dict
             suboptions:
               name:
                 description: Name of the area (eg Area1).
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:55]: updated.

File: plugins/modules/site_intent.py [hunk 14]
```diff
@@ -35,50 +35,48 @@
...
                 type: str
-              parentName:
-                description: Complete Parent name of the Area to be created/deleted(eg Global/).
+              parent_name:
+                description: Complete Parent name of the Area to be created/deleted(eg Global/USA).
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:58]: updated.

File: plugins/modules/site_intent.py [hunk 15]
```diff
@@ -35,50 +35,48 @@
...
                 type: str
           building:
-            description: Building Details.
+            description: Contains details for creating or managing a building within a site.
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:61]: updated.

File: plugins/modules/site_intent.py [hunk 16]
```diff
@@ -35,50 +35,48 @@
...
             type: dict
             suboptions:
               address:
                 description: Address of the building to be created.
```
[the remark this answers is not in the document]
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:65]: updated.

File: plugins/modules/site_intent.py (hunk 1 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:68]: added the data type as float.

File: plugins/modules/site_intent.py (hunk 3 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:73]: updated.

File: plugins/modules/site_intent.py (hunk 5 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:79]: updated and added the data type as float.

File: plugins/modules/site_intent.py (hunk 7 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:95]: updated.

File: plugins/modules/site_intent.py (hunk 9 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:111]: updated.

File: plugins/modules/site_intent.py (hunk 6 above)
Comment (Abhishek-121) [reply plugins/modules/site_intent.py:92]: updated.

File: plugins/modules/site_intent.py [hunk 17]
```diff
@@ -35,73 +35,87 @@
...
-    - List of details of site being managed.
+    description: It represents a list of details for creating/managing/deleting sites, including areas, buildings, and floors.
     type: list
     elements: dict
-    required: true
+    required: True
     suboptions:
-      type:
+      site_type:
         description: Type of site to create/update/delete (eg area, building, floor).
```
Comment (madhansansel) [inline plugins/modules/site_intent.py:44]: description: Type of site to create/update/delete (e.g., area, building, floor).

**Review by madhansansel — APPROVED** [review]

**Pull request merged**